lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,39 +9,66 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache.
+	 * It can be cleared by calling wp_theme_has_theme_json_clean_cache().
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
+	function wp_theme_has_theme_json() {
+		$cache_group       = 'theme_json';
+		$cache_key         = 'wp_theme_has_theme_json';
+		$theme_has_support = wp_cache_get( $cache_key, $cache_group );
 
-		if ( true === $clear_cache ) {
-			$theme_has_support = null;
-		}
-
-		if ( null !== $theme_has_support ) {
-			return $theme_has_support;
+		/**
+		 * $theme_has_support is stored as a int in the cache.
+		 *
+		 * The reason not to store it as a boolean is to avoid working
+		 * with the $found parameter which apparently had some issues in some implementations
+		 * https://developer.wordpress.org/reference/functions/wp_cache_get/
+		 */
+		if ( ( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
+			( 0 === $theme_has_support || 1 === $theme_has_support ) ) {
+			return (bool) $theme_has_support;
 		}
```

Comment:
Changing behavour based on debug flags is not a great idea. It can make debugging issues really hard.

Reply:
There are different [views](https://github.com/WordPress/gutenberg/pull/45543#discussion_r1014284176) on how/when to use `WP_DEBUG`. I'm going to remove it from this PR to unblock it. It can be revisited later in a separate PR should we need.

Reply:
@spacedmonkey The point of `WP_DEBUG` here is that we shouldn't use a cache for data controlled by code when in "development" mode. If we don't add `WP_DEBUG` here, this is going to be extremely painful to work with e.g. as a theme developer who is working on their `theme.json` and then will never see their changes reflected.

Usually I would agree with what you're saying here about `WP_DEBUG`, but there is a difference in how we need to handle caching data from the DB vs caching information that is purely controlled by code.

Reply:
https://github.com/WordPress/gutenberg/pull/45882 brings back this conversation, for anyone interested in following up.